brave/brave-core
# (Wallet) Improve biometric authentication using cipher

## Resolves https://github.com/brave/brave-browser/issues/40188

Improve authentication security during onboarding process and unlock by using cipher from biometric authentication.

🔍 Security review required: https://github.com/brave/reviews/issues/1727
✅ Change base branch to `master` once https://github.com/brave/brave-core/pull/24943 is merged.




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Clean Wallet data, is any
- Navigate to Wallet section to create a new Wallet
- Enable fingerprint unlock and complete onboarding process
- Lock Wallet
- Unlock Wallet using fingerprint unlock
- Observe the Wallet is unlock correctly
